src/backend/distributed/metadata/node_metadata.c

```diff
@@ -787,7 +787,6 @@ SyncDistributedObjectsToNode(WorkerNode *workerNode)
 		return;
 	}
 
-	EnsureNoModificationsHaveBeenDone();
```

Comment:
As noted in the commit message, we already have `EnsureSequentialModeMetadataOperations`, and the whole operation is always transactional. So `EnsureNoModificationsHaveBeenDone()` is too strict. 

For example, #5556 fails because of that